File: java/src/main/java/com/techelevator/Services/ReviewService.java
package com.techelevator.Services;

import com.techelevator.dao.ReviewDao;
import com.techelevator.model.Review;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ReviewService {

    private ReviewDao reviewDao;
    @Autowired
    public ReviewService(ReviewDao reviewDao) {
        this.reviewDao = reviewDao;
    }
    public List<Review> getAllReviews() {
        return reviewDao.getAllReviews();
    }
    public Review getReview(int review_id) {
        return reviewDao.getReview(review_id);
    }
    public Review addReview(Review newReview) {
        return reviewDao.addReview(newReview);
    }
    public Review updateReview(Review updatedReview) {
        return reviewDao.updateReview(updatedReview);
    }
    public int deleteReview(int review_id) {
        return reviewDao.deleteReview(review_id);
    }
}

File: vue/src/services/ReviewService.js
import axios from 'axios';

export default {

    getAllReviews() {
        return axios.get(`/allReviews`);
    },
    getReview(review_id) {
        return axios.get(`/reviews/${review_id}`)
    },
    addReview(newReview) {
        return axios.post('/addReview', newReview)
    },
    updateReview(updatedReview, review_id){
        return axios.put(`/reviews/${review_id}`, updatedReview)
    },
    deleteReview(review_id){
        return axios.delete(`breweries/${review_id}`)
    }
}


 